Source/msg.h

```diff
@@ -135,98 +135,62 @@ struct TCmd {
 	_cmd_id bCmd;
 };
 
-struct TCmdLoc {
-	_cmd_id bCmd;
+struct TCmdLoc : TCmd {
 	uint8_t x;
 	uint8_t y;
 };
 
-struct TCmdLocParam1 {
-	_cmd_id bCmd;
-	uint8_t x;
-	uint8_t y;
+struct TCmdLocParam1 : TCmdLoc {
 	uint16_t wParam1;
 };
 
-struct TCmdLocParam2 {
-	_cmd_id bCmd;
-	uint8_t x;
-	uint8_t y;
-	uint16_t wParam1;
+struct TCmdLocParam2 : TCmdLocParam1 {
 	uint16_t wParam2;
 };
 
-struct TCmdLocParam3 {
-	_cmd_id bCmd;
-	uint8_t x;
-	uint8_t y;
-	uint16_t wParam1;
-	uint16_t wParam2;
+struct TCmdLocParam3 : TCmdLocParam2 {
 	uint16_t wParam3;
 };
 
-struct TCmdLocParam4 {
-	_cmd_id bCmd;
-	uint8_t x;
-	uint8_t y;
-	uint16_t wParam1;
-	uint16_t wParam2;
-	uint16_t wParam3;
+struct TCmdLocParam4 : TCmdLocParam3 {
 	uint16_t wParam4;
 };
 
-struct TCmdParam1 {
-	_cmd_id bCmd;
+struct TCmdParam1 : TCmd {
 	uint16_t wParam1;
 };
 
-struct TCmdParam2 {
-	_cmd_id bCmd;
-	uint16_t wParam1;
+struct TCmdParam2 : TCmdParam1 {
 	uint16_t wParam2;
 };
 
-struct TCmdParam3 {
-	_cmd_id bCmd;
-	uint16_t wParam1;
-	uint16_t wParam2;
+struct TCmdParam3 : TCmdParam2 {
 	uint16_t wParam3;
 };
 
-struct TCmdParam4 {
-	_cmd_id bCmd;
-	uint16_t wParam1;
-	uint16_t wParam2;
-	uint16_t wParam3;
+struct TCmdParam4 : TCmdParam3 {
 	uint16_t wParam4;
 };
 
-struct TCmdGolem {
-	_cmd_id bCmd;
-	uint8_t _mx;
-	uint8_t _my;
+struct TCmdGolem : TCmdLoc {

```

Comment:
There are no guarantees about the memory layout here because inherited structs are no longer POD.
We have to be very careful with how we unpack these. Definitely can't just cast them from a network byte stream.

Reply:
Sounds like it's best to just close this PR and leave it as is :)